core/arch/arm/plat-vexpress/platform_config.h

```diff
@@ -227,7 +227,7 @@
 #define DRAM0_BASE		UINTPTR_C(0x40000000)
 #define DRAM0_SIZE		(UINTPTR_C(0x40000000) - CFG_SHMEM_SIZE)
 
-#define DRAM0_TEERES_BASE	(DRAM0_BASE + DRAM0_SIZE)
+#define DRAM0_TEERES_BASE	(DRAM0_BASE + DRAM0_SIZE - CFG_SHMEM_SIZE)
```

Comment:
Subtracting `CFG_SHMEM_SIZE` here doesn't make sense. Also `DRAM0_TEERES_*` has outlived its purpose. I'd rather hard code the value for `CFG_SHMEM_START` directly with a comment telling why this spot in memory is a good spot.